app/models/external_activity.rb

```diff
@@ -165,7 +165,17 @@ def url(learner = nil)
     end
   end
 
+  # 
+  # external_type: "Investigation", "activity", "Interactive"
+  #
+  # New ExternalActivities (ext_type interactives) appear in Interactives bin.
+  #
+  # interactives_controller searched old interactives
+  #
   def material_type
+    if external_type && !external_type.blank?
+        return external_type
+    end
```

Comment:
As I look at this as a whole, I think it might be better to:
- name the field `material_type`
- remove this `material_type` method
- create a migration that sets the `material_type` for all existing external activities. It would set it to template_type, if there is a template, and Activity if not.
- update the code that creates ExternalActivities when a LARA material is published, so the material_type is set to the template_type.

What do you think?

Reply:
> update the code that creates ExternalActivities when a LARA material is published, so the material_type is set to the template_type.

@scytacki Can you point me at where I look to find this code? I'm guessing this is the publish activity path when a user publishes from LARA to portal? Does this require changes on the LARA side? (I would guess no.)